javascripts/googleAnalyticsImporter.js

```diff
@@ -0,0 +1,91 @@
+(function () {
+  document.addEventListener("DOMContentLoaded", function(event) {
+    window.CoreHome.Matomo.on("matomoPageChange", checkForPendingImporters);
+    window.CoreHome.Matomo.on("piwikPageChange", checkForPendingImporters);
+  });
+  checkForPendingImporters();
+  function checkForPendingImporters(){
+    let params = getAllUrlParams(window.location.href);
+
+    if(!params.hasOwnProperty("date") || !params.hasOwnProperty("period") || !params.hasOwnProperty("idsite")){
+      return;
+    }
+
+    let searchParams = {
+      idSite : params.idsite,
+      date : params.date,
+      period : params.period,
+      module: "GoogleAnalyticsImporter",
+      action: "pendingImports"
+    };
+
+    //?module=CoreAdminHome&action=home&idSite=1&period=day&date=yesterday";
+    (async () => {
+      const response = await fetch('/index.php?' + new URLSearchParams(searchParams));
+      const data = await response.json();
+      if (data.displayPending){
+        displayPendingNotification();
+      } else {
+        hidePendingNotification();
+      }
+    })();
+  }
+})();
+
+function hidePendingNotification(){
+  var UI = require('piwik/UI');
+  var notification = new UI.Notification();
+  notification.remove('GoogleAnalyticsImporterPendingImportNotice');
+}
+
+
+function displayPendingNotification(){
+  var UI = require('piwik/UI');
+  var notification = new UI.Notification();
+  notification.show(_pk_translate("GoogleAnalyticsImporter_PendingGAImportReportNotification"), {
+    context: 'info',
+    noclear: false,
+    type: 'toast',
+    id: 'GoogleAnalyticsImporterPendingImportNotice'
+  });
+}
+
+function getAllUrlParams(url) {
```

Comment:
I don't see a function for getting all parameters, but it looks like there's already a Matomo.getUrlParameter() function, if you already know which parameter you're looking for.

Reply:
Hi @snake14, it should be in line 8 in the same file.

Reply:
Sorry. I didn't word my comment very clearly. I was just questioning whether you needed to define a `getAllUrlParams` function when there is already a `Matomo.getUrlParameter()` function. It's not a big deal though.